packages/components/src/DataList/DataList.tsx

```diff
@@ -24,14 +27,25 @@ export function DataList<T extends DataListObject>({
   title,
   totalCount,
 }: DataListProps<T>) {
-  const layout = getCompoundComponent<DataListLayoutProps<T>>(
+  const allLayouts = getCompoundComponents<DataListLayoutProps<T>>(
     children,
     DataListLayout,
-  )?.props.children;
+  );
 
   const elementData = generateListItemElements(data);
   const headerData = generateHeaderElements(headers);
+  const mediaMatches = useGridLayoutMediaQueries();
 
+  const dataListHeaderContent = renderDataListHeader(
+    allLayouts,
+    headerData,
+    mediaMatches,
+  );
+  const dataListContent = renderDataListItems(
+    allLayouts,
+    elementData,
+    mediaMatches,
+  );
```

Comment:
I wonder if these things should be encapsulated in its own component now rather than functions that returns an element.

Basically separate concerns.
- `<DataListHeader />` only deals with header bits
- `<DataListItems />` only deals with the items
- Both of them access' the media matches